src/test/java/org/example/springintro/services/CategoryServiceTest.java

```diff
@@ -0,0 +1,197 @@
+package org.example.springintro.services;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+import org.example.springintro.dto.category.CategoryDto;
+import org.example.springintro.dto.category.CreateCategoryRequestDto;
+import org.example.springintro.exception.EntityNotFoundException;
+import org.example.springintro.mapper.CategoryMapper;
+import org.example.springintro.model.Category;
+import org.example.springintro.repository.categoty.CategoryRepository;
+import org.example.springintro.services.impl.CategoryServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceTest {
```

Comment:
the same about using TestUtils. Also, you can use Spy for mappers and simplify your test cases ;)